libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1807,13 +1807,18 @@ bool TypeChecker::visit(FunctionCall const& _functionCall)
 			argumentsArePure = false;
 	}
 
-	// For positional calls only, store argument types
-	if (_functionCall.names().empty())
+	// Store argument types - and names if given - for overload resolution
 	{
-		shared_ptr<TypePointers> argumentTypes = make_shared<TypePointers>();
+		FuncCallArguments funcCallArgs;
+
+		if (!_functionCall.names().empty())
```

Comment:
Can't you replace all three of these lines by just `funCallArgs.names = _funCall.names()`?